CC/Sounder/config.cc

```diff
@@ -265,20 +265,16 @@ Config::Config(const std::string& jsonfile)
     // 15reps of STS(16) + 2reps of gold_ifft(128)
     srand(time(NULL));
     const int seqLen = 128;
-    std::vector<std::vector<double>> gold_ifft
-        = CommsLib::getSequence(CommsLib::GOLD_IFFT);
-    std::vector<std::complex<int16_t>> gold_ifft_ci16
-        = Utils::double_to_cint16(gold_ifft);
+    auto gold_ifft = CommsLib::getSequence(CommsLib::GOLD_IFFT);
+    auto gold_ifft_ci16 = Utils::float_to_cint16(gold_ifft);
     gold_cf32_.clear();
     for (size_t i = 0; i < seqLen; i++) {
         gold_cf32_.push_back(
             std::complex<float>(gold_ifft[0][i], gold_ifft[1][i]));
     }
 
-    std::vector<std::vector<double>> sts_seq
-        = CommsLib::getSequence(CommsLib::STS_SEQ);
-    std::vector<std::complex<int16_t>> sts_seq_ci16
-        = Utils::double_to_cint16(sts_seq);
+    auto sts_seq = CommsLib::getSequence(CommsLib::STS_SEQ);
```

Comment:
General statement:  no action.  Try to only use 'auto' when the type is obvious from the statement or trivial.   It isn't immediately obvious of the return type CommsLib::getSequence us a vector of complex values.

"The fundamental rule is: use type deduction only to make the code clearer or safer, and do not use it merely to avoid the inconvenience of writing an explicit type. When judging whether the code is clearer, keep in mind that your readers are not necessarily on your team, or familiar with your project, so types that you and your reviewer experience as unnecessary clutter will very often provide useful information to others. For example, you can assume that the return type of make_unique<Foo>() is obvious, but the return type of MyWidgetFactory() probably isn't.

These principles apply to all forms of type deduction, but the details vary, as described in the following sections."

Reply:
Great advice! Thanks.